frontend/packages/kubevirt-plugin/src/components/modals/menu-actions-modals/delete-vmi-modal.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { k8sKill } from '@console/internal/module/k8s';
+import { HandlePromiseProps, history, withHandlePromise } from '@console/internal/components/utils';
+import { YellowExclamationTriangleIcon } from '@console/shared/src/components/status/icons';
+import { getName, getNamespace } from '@console/shared/src/selectors/common';
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { VMIKind } from '../../../types';
+import { VirtualMachineInstanceModel } from '../../../models';
+
+const redirectFn = (vmi: VMIKind) => {
+  // If we are currently on the deleted resource's page, redirect to the resource list page
+  const re = new RegExp(`/${getName(vmi)}(/|$)`);
+  if (re.test(window.location.pathname)) {
+    history.push(`/k8s/ns/${getNamespace(vmi)}/virtualization`);
+  }
+};
+
+export const DeleteVMIModal = withHandlePromise((props: DeleteVMIProps) => {
```

Comment:
Should we also ask for disk removal in standalone VMIs which do not belong to any VM resource?

Reply:
added :-)